kernel/shared/src/main/scala/monocle/function/Reverse.scala

```diff
@@ -0,0 +1,48 @@
+package monocle.function
+
+import monocle.Iso
+
+trait Reverse[A] {
+  type B
+
+  def reverse: Iso[A, B]
+}
+
+object Reverse {
+  type Aux[A, B0] = Reverse[A] { type B = B0 }
```

Comment:
maybe worth to have a second type alias for the case where `A = B`

Reply:
do you mean
`type Aux[A] = Reverse[A] {type B = A}`?
I guess it would need another name

Reply:
yes, unfortunately, I don't have a good suggestion for names.